rafiki/model/dev.py

```diff
@@ -104,18 +108,28 @@ def tune_model(py_model_class: Type[BaseModel], train_dataset_path: str, val_dat
         store_params_id = _save_model(model_inst, proposal, result, param_cache, param_store)
 
         # Update best saved model
-        if result.score is not None and store_params_id is not None and result.score > best_model_score:
-            inform_user('Best saved model so far! Beats previous best of score {}!'.format(best_model_score))
-            best_store_params_id = store_params_id
-            best_proposal = proposal
-            best_model_score = result.score
-            best_trial_no = trial_no
-
-            # Test best model, if test dataset provided
-            if test_dataset_path is not None:
-                print('Evaluting new best model on test dataset...')
-                best_model_test_score = model_inst.evaluate(test_dataset_path)
-                inform_user('Score on test dataset: {}'.format(best_model_test_score))
+        if result.score is not None and store_params_id is not None:
+            if best_model_score == 0:
+                is_best_model = True
+            else:
+                is_best_model = (result.score < best_model_score) if train_args['task_type'] == 'regression' \
+                    else (result.score > best_model_score)
+            if is_best_model:
+                inform_user('Best saved model so far! Beats previous best of score {}!'.format(best_model_score))
+                best_store_params_id = store_params_id
+                best_proposal = proposal
+                best_model_score = result.score
+                best_trial_no = trial_no
+
+                # Need to minimize RMSE
+                if train_args['task_type'] == 'regression':
```

Comment:
Editing the tuning logic here is simply edits it for *local development* by model developers, but doesn't update the logic for model training workers that are actually deployed on Rafiki. Moreover, I believe we should do the inversion of the RMSE *within the `evaluate` method* of the model to avoid special-casing like this - and it conforms to the [method's description](https://nginyc.github.io/rafiki/docs/0.2.0/src/python/rafiki.model.html#rafiki.model.BaseModel.evaluate)

Reply:
I agree with your opinion on inverting the RMSE for the consistency of the `evaluate` method.